debugProtocol.json

```diff
@@ -3190,6 +3359,50 @@
 					"description": "Details of the exception contained by this exception, if any."
 				}
 			}
+		},
+
+		"DisassembledInstruction": {
+			"type": "object",
+			"description": "Represents a single disassembled instruction.",
+			"properties": {
+				"address": {
+					"type": "string",
+					"description": "The address of the instruction.  Treated as a hex value if prefixed with '0x', or as a decimal value otherwise."
+				},
+				"instructionBytes": {
+					"type": "string",
+					"description": "Raw bytes representing the instruction and its operands, in an implementation-defined format."
+				},
+				"instruction": {
+					"type": "string",
+					"description": "Text representing the instruction and its operands, in an implementation-defined format."
+				},
+				"symbol": {
```

Comment:
More of a question. Does it strictly have to be an actual symbol or can it be something like `<main+0x52>`

Reply:
I believe the intent here is to display things like goto labels. So you probably wouldn't want to return something like `<main+0x52>` as that would presumably stick a label on every instruction. That said, I would say that a proper client UI shouldn't try and interpret `symbol` at all. So if a debug adapter thought that users would like to see `<main+0x52>` then that would be totally acceptable.

Reply:
Sorry, I was wrong about symbols. No such thing as `<main+52>` as a symbol. I was more worried about what happens after the instruction. Some examples below (the first group is an actual symbol the others are just (helpful) annotation which is sometimes as a comment and others as not. @andrewcrawley [confirmed](https://github.com/microsoft/vscode-debugadapter-node/pull/212#discussion_r287889870) that other than address, the DA can do anything with other fields.
```
0x10080edc <main>:
0x10080edc:	b480      	push	{r7}

0x10080ef0:	4b07      	ldr	r3, [pc, #28]	; (10080f10 <main+0x34>)
0x10080f0c:	e7ef      	b.n	10080eee <main+0x12>
0x10002590: 03 4b           	ldr	r3, [pc, #12]	; (0x100025a0 <main+24>)
0x1000259a: 01 f0 fd f8     	bl	0x10003798 <Cy_SysLib_Delay>
0x1000259e: f7 e7           	b.n	0x10002590 <main+8>
```
Upto the client to dicide how to display all of this, but it is the DA job to consolidate the first two lines above into one `DisassembledInstruction` because there is no null instruction?

Reply:
> Upto the client to dicide how to display all of this, but it is the DA job to consolidate the first two lines above into one `DisassembledInstruction` because there is no null instruction?
Correct